### PR TITLE
feat: [CDS-70142]: bug fixes for overriddes 2.0 post testing

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
### Summary

Bug fixes for overrides 2.0 post testing

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/interfaces/SecretsInterface.ts [hunk 1]
```diff
@@ -16,3 +18,9 @@ export enum PrincipalScope {
   ORG = 'organization',
   ACCOUNT = 'account'
 }
+
+export const scopeStringKey: Record<Scope, StringKeys> = {
+  account: 'account',
+  org: 'orgLabel',
```
Comment (vikrant-harness) [inline src/modules/10-common/interfaces/SecretsInterface.ts:24]: orgLabel ? is this correct ?

File: src/modules/75-cd/components/ServiceOverrides/ServiceOverridesUtils.ts [hunk 2]
```diff
@@ -47,132 +48,157 @@ export const noOverridesStringMap: Record<Required<ServiceOverridesResponseDTOV2
...
-      width: 150,
+      headerWidth: 150,
+      rowWidth: 150,
       value: 'common.serviceOverrides.overrideType',
       accessKey: 'overrideType',
       mapper: overridesLabelStringMap
     },
     {
-      width: '73%',
+      // headerWidth: '73%',
```
Comment (vikrant-harness) [inline src/modules/75-cd/components/ServiceOverrides/ServiceOverridesUtils.ts:67]: commented code ?

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/10-common/interfaces/SecretsInterface.ts (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/10-common/interfaces/SecretsInterface.ts:24]: Yes

Comment (AshwinBhatkal-Harness) [reply src/modules/10-common/interfaces/SecretsInterface.ts:24]: That is the string key

File: src/modules/75-cd/components/ServiceOverrides/ServiceOverridesUtils.ts (hunk 2 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/75-cd/components/ServiceOverrides/ServiceOverridesUtils.ts]: Removed

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/RowItemFromValue/ScopedEntitySelect/EnvironmentEntityList.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        !isEmpty(listData) ? listData.map(entityInList => entityInList.environment as EnvironmentResponseDTO) : []
+      )
+    }
+  }, [loadingList, listResponse])
+
+  useEffect(() => {
+    /* istanbul ignore else */
+    if (listError?.message) {
+      /* istanbul ignore else */
+      if (shouldShowError(listError)) {
```
Comment (YounixIO) [inline src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/RowItemFromValue/ScopedEntitySelect/EnvironmentEntityList.tsx:72]: Merged the if statements in future PR

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useConnectionStringOverride.tsx [hunk 4]
```diff
@@ -82,6 +85,14 @@ export default function useConnectionStringOverride({
   const { repoIdentifier, branch } = useQueryParams<GitQueryParams>()
   const { expressions } = useVariablesExpression()
 
+  const editConnectionString = (): void => {
+    if (connectionStrings?.store.type) {
```
Comment (YounixIO) [inline src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useConnectionStringOverride.tsx:89]: Add a type safety check

File: src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx [hunk 5]
```diff
@@ -136,14 +136,10 @@ export default function useManifestOverride({
...
-  const editManifestOverride = (
-    manifestType: OverrideManifestTypes,
-    store: OverrideManifestStoresTypes,
-    index: number
-  ): void => {
+  const editManifestOverride = (manifestType: OverrideManifestTypes, store: OverrideManifestStoresTypes): void => {
     setSelectedManifest(manifestType)
     setManifestStore(store)
-    setEditIndex(index)
+    setEditIndex(0)
```
Comment (YounixIO) [inline src/modules/75-cd/components/ServiceOverrides/components/ListRows/Editable/useManifestOverride.tsx:142]: Add a comment above the code change mentioning the details of choosing `0` in place of index.

**Review by YounixIO — APPROVED** [review]

**Pull request merged**